homeassistant/components/shelly/switch.py

```diff
@@ -291,8 +267,8 @@ def is_on(self) -> bool:
 
     async def async_turn_on(self, **kwargs: Any) -> None:
         """Turn on relay."""
-        await self.call_rpc("Switch.Set", {"id": self._id, "on": True})
+        await self.call_rpc("Switch.Set", {"id": self.status["id"], "on": True})
 
     async def async_turn_off(self, **kwargs: Any) -> None:
         """Turn off relay."""
-        await self.call_rpc("Switch.Set", {"id": self._id, "on": False})
+        await self.call_rpc("Switch.Set", {"id": self.status["id"], "on": False})
```

Comment:
Is this change needed? Can we keep the `_id` since we kept `__init__`?

Reply:
Done